### ISSUE TITLE
Build Fails on CentOS 7

### ISSUE THREAD

**Issue by blackknight36** [bisqwit/nescom#1]
Is there a way to compile this code on CentOS 7?  Running *make* results in an error as follows.

> g++: error: unrecognized command line option ‘-std=c++14’

**Comment by bisqwit** [comment]
You need a compiler that is not older than 4 years.
Suitable compilers:
— GCC version 4.9 or more recent (released in 2014-04-22)
— Clang version 3.5 or more recent (released in 2014-09-03)
— ICC version 15.0 or more recent (released in 2014-07-25)

**Comment by blackknight36** [comment]
Thanks.  I guess I'll have to try building on a different system.

**Comment by blackknight36** [comment]
Looks like there's an issue compiling on Fedora 28 as well.  Using gcc version 8.1.1 shows an error as follows.

```
g++ -std=c++14 -Ofast -Wall -Wundef -Wcast-qual -Wpointer-arith -Wconversion -Wwrite-strings -Wsign-compare -Wredundant-decls -Winit-self -Wextra -Wcast-align -Wformat -Woverloaded-virtual -Wno-conversion -g -o disasm disasm.o romaddr.o o65.o
disasm.o: In function `LoadO65globals(O65 const&, SegmentSelection)':
/home/00/d861703/Dropbox/games/nescom/disasm.cc:213: undefined reference to `O65::GetSymbolList(SegmentSelection) const'
/home/00/d861703/Dropbox/games/nescom/disasm.cc:217: undefined reference to `O65::GetSymAddress(SegmentSelection, std::string const&) const'
disasm.o: In function `HandleO65':
/home/00/d861703/Dropbox/games/nescom/disasm.cc:250: undefined reference to `O65::GetExternList() const'
collect2: error: ld returned 1 exit status
make: *** [Makefile:118: disasm] Error 1
```

**Comment by bisqwit** [comment]
Did you do 'make clean' first?

Note: nescom also fails on a too *recent* compiler (though not for the reason quoted above). It needs to be updated some day. I already did the changes to snescom a few weeks ago, but nescom is a little behind.

**Comment by blackknight36** [comment]
I ran make clean and make which now fails with a different error.

```
link.cc: In function ‘void MessageLoadingItem(const string&)’:
link.cc:52:39: warning: unused parameter ‘header’ [-Wunused-parameter]
 void MessageLoadingItem(const string& header)
                         ~~~~~~~~~~~~~~^~~~~~
In file included from /usr/include/c++/8/map:61,
                 from dataarea.hh:4,
                 from link.cc:11:
/usr/include/c++/8/bits/stl_map.h: In instantiation of ‘class std::map<unsigned int, rangeset<unsigned int>::Valueholder, std::less<unsigned int>, std::allocator<unsigned int> >’:
range.hh:49:10:   required from ‘class rangecollection<unsigned int, rangeset<unsigned int>::Valueholder, std::allocator<unsigned int> >’
rangeset.hh:28:10:   required from ‘class rangeset<unsigned int>’
/usr/include/c++/8/bits/stl_pair.h:204:11:   required from ‘struct std::pair<const unsigned int, rangeset<unsigned int> >’
/usr/include/c++/8/ext/aligned_buffer.h:54:65:   required from ‘struct __gnu_cxx::__aligned_membuf<std::pair<const unsigned int, rangeset<unsigned int> > >’
/usr/include/c++/8/bits/stl_tree.h:231:41:   required from ‘struct std::_Rb_tree_node<std::pair<const unsigned int, rangeset<unsigned int> > >’
/usr/include/c++/8/bits/stl_tree.h:1872:21:   required from ‘void std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_erase(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Link_type) [with _Key = unsigned int; _Val = std::pair<const unsigned int, rangeset<unsigned int> >; _KeyOfValue = std::_Select1st<std::pair<const unsigned int, rangeset<unsigned int> > >; _Compare = std::less<unsigned int>; _Alloc = std::allocator<std::pair<const unsigned int, rangeset<unsigned int> > >; std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_Link_type = std::_Rb_tree_node<std::pair<const unsigned int, rangeset<unsigned int> > >*]’
/usr/include/c++/8/bits/stl_tree.h:965:9:   required from ‘std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::~_Rb_tree() [with _Key = unsigned int; _Val = std::pair<const unsigned int, rangeset<unsigned int> >; _KeyOfValue = std::_Select1st<std::pair<const unsigned int, rangeset<unsigned int> > >; _Compare = std::less<unsigned int>; _Alloc = std::allocator<std::pair<const unsigned int, rangeset<unsigned int> > >]’
/usr/include/c++/8/bits/stl_map.h:300:7:   required from here
/usr/include/c++/8/bits/stl_map.h:122:21: error: static assertion failed: std::map must have the same value_type as its allocator
       static_assert(is_same<typename _Alloc::value_type, value_type>::value,
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [<builtin>: link.o] Error 1
```

This is happening on Fedora 28 with GCC version 8.1.1.

**Comment by bisqwit** [comment]
Indeed. That is the problem I *was* referring to in my previous comment. Known problem, and I am sorry. I have to address that some day. For now, try a slightly older compiler. Again, sorry about that.

Or, as a quick fix, edit `range.hh` and find the line that has `//typename` on it. Remove the `//` and add the same `//` to the beginning of the very next line. It should now compile fully, but I have not tested yet if it runs as intended with this minimal fix.

**Comment by bisqwit** [comment]
Fixed in release version 1.2.0.

**Comment by blackknight36** [comment]
@bisqwit Glad that it's fixed.
